added analytics to studypact

Kwang8 opened the pull request:
Added analytics to CreatePreference, ReviewPreference, Profile, PendingGroupInvite, AllGroups, and the GroupDetailView. Not added to the Firebase yet, so I don't know how to check if it works.

huangshawn1 commented on berkeley-mobile/StudyPact/CreatePreference/ReviewPreferencesViewController.swift:
```diff
@@ -50,6 +51,7 @@ class ReviewPreferencesViewController: UIViewController {
     
     override func viewWillAppear(_ animated: Bool) {
         updateLabels()
+        Analytics.logEvent("opened_studypact_reviewPreferences", parameters: nil)
```
change to use underscores instead of camelcase (opened_studypact_review_preferences)

huangshawn1 replied:
same for the rest

huangshawn1 reviewed (commented):
looks good, could you also add analytics events for the academic calendar events and campus-wide events pages?

Kwang8 commented:
Changed all the camel cases to underscores, also added analytics to the academic calendar and campus events. Sorry for all the commits, forgot to delete prints.
